### PR TITLE
Add test CI action

### PR DESCRIPTION
**Pull request by nathanshelly**

### CODE REVIEW

**Comment by sashaweiss** [comment]
no tests pls add tests

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build
+      run: cargo build --verbose
+    - name: Run tests
+      run: cargo test --verbose
```
Comment (sashaweiss) [inline .github/workflows/build_and_test.yml:13]: there are no tests pls add tests

**Review by sashaweiss — CHANGES_REQUESTED** [review]

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (nathanshelly) [reply .github/workflows/build_and_test.yml:13]: tests added

File: src/snek/foo.rs [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
-    println!("Hello world!");
+  println!("Hello world!");
+}
+#[cfg(test)]
+mod tests {
+  use super::*;
+  #[test]
+  fn panic() {
+    foo();
+    panic!();
```
Comment (sashaweiss) [inline src/snek/foo.rs]: test is failing pls make test not fail

**Review by sashaweiss — CHANGES_REQUESTED** [review]

**Review by sashaweiss — APPROVED** [review]
lgtm

**Comment by sashaweiss** [comment]
@nathanshelly  prefer this as a squash merge, if that's kosher

**Pull request merged**